packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -959,8 +962,11 @@ export const DatePickerMixin = (subclass) =>
      */
     _onClearButtonClick(event) {
       event.preventDefault();
-      this.value = '';
       this._inputElementValue = '';
```

Comment:
The order in which we clear the component's value and the input element's value isn't important. The only thing that matters is that `_hasInputValue` is reset before the component's value.